code/src/components/Labyrinth.js

```diff
@@ -0,0 +1,32 @@
+import React from "react";
+import { useSelector } from "react-redux";
+import GameOver from "./GameOver";
+import StartPage from './StartPage';
+import GameScreen from "./GameScreen";
+import Loading from "./Loading";
+
+import { Container } from "./Styles";
+
+const Labyrinth = () => {
+    const stage = useSelector(store => store.labyrinth.stage);
+    const isLoading = useSelector(store => store.labyrinth.isLoading);
+
+    return (
+        <>
+            {isLoading && <Loading />}
+            {!isLoading && (
+                <Container>
+                    {stage 
+                        ? stage.actions.length !== 0
+                            ? <GameScreen />
+                            : <GameOver />
+                        : <StartPage/> 
+                    }
+                </Container>

```

Comment:
Really nice use of a nested ternary operator to render all three different components! Really cool to see in use. I also like how you guys wrapped all the components in the Container from Styles.